packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,140 @@
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  for (const item of items) {
+    // only rename the file that has same name with component
+    if (isNameMatch(item, componentName, componentPath)) {
+      const newItem = item.replace(componentName, newName);
+      fs.renameSync(
+        path.join(componentPath, item),
+        path.join(componentPath, newItem)
+      );
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath
+  );
+  notificationService.showWarningMessage(nls.localize(RENAME_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    const errorMessage = nls.localize(RENAME_INPUT_DUP_ERROR);
+    notificationService.showErrorMessage(errorMessage);
+    throw new Error(format(errorMessage));
+  }
+}
+
+function isDuplicate(componentPath: string, newName: string): boolean {
+  // A LWC component can't share the same name as a Aura component
+  const componentPathDirName = path.dirname(componentPath);
+  let lwcPath: string;
+  let auraPath: string;
+  if (isLwcComponent(componentPath)) {
+    lwcPath = componentPathDirName;
+    auraPath = path.join(path.dirname(componentPathDirName), 'aura');
+  } else {
+    lwcPath = path.join(path.dirname(componentPathDirName), 'lwc');
+    auraPath = componentPathDirName;
+  }
+  const allLwcComponents = fs.readdirSync(lwcPath);
+  const allAuraComponents = fs.readdirSync(auraPath);
+  if (allLwcComponents.includes(newName) || allAuraComponents.includes(newName)) {
+    return true;
+  }
+  return false;
+}
+
+function isNameMatch(item: string, componentName: string, componentPath: string) {
+  const isLwc = isLwcComponent(componentPath);
+  let regularExp: RegExp;
+  if (isLwc) {
+    regularExp = new RegExp(`${componentName}\.(html|js|js-meta.xml|css|svg)`);
+  } else {
+    regularExp = new RegExp(`${componentName}(((Controller|Renderer|Helper)?\.js)|(\.(cmp|app|css|design|auradoc|svg)))`);
+  }
+  return item.match(regularExp) ? true : false;
+}
+
+function isLwcComponent(componentPath: string): boolean {
+  return path.basename(path.dirname(componentPath)) === 'lwc' ? true : false;
```

Comment:
another place to use that 'lwc' constant

Reply:
I don't quite understand why here needs 'lwc' constant.

Reply:
It's def a person preference but having the string 'lwc' spread though out our code base can just open us up to typo errors.   I prefer to have a constants file somewhere where we have 
`export const LWC = 'lwc';` that we reference everywhere instead of have the string 'lwc'.  If we only use it in this file then having it just defined in the file is fine.  The most obviously benefit outside of just us have to avoid the mental tracking of when 'lwc' should be use is in tests.  It's much better to use a constant LWC in the source, then verify in the test that LWC was used.